eZ/Publish/Core/Repository/PermissionsCriterionHandler.php

```diff
@@ -118,6 +118,7 @@ public function getPermissionsCriterion($module = 'content', $function = 'read')
             foreach ($permissionSet['policies'] as $policy) {
                 $limitations = $policy->getLimitations();
                 if ($limitations === '*' || empty($limitations)) {
+                    $policyOrCriteria[] = new Criterion\MatchAll();
```

Comment:
I think I agree, it should be optimized on store. But for the sake of fixing the bug now, please check if what @pspanja suggests passes integration tests and go with that here.

But do add a comment about it, like
```php
// Given policy gives full access, optimize away this whole role as it does not need to filter the query
$policyOrCriteria = [];
break;
```